### PR TITLE
Ansible setup

### PR DESCRIPTION
**Pull request by tpenning**
Create the Ansible playbooks that create the Kubernetes cluster with a control node and worker nodes.
Install all frameworks required that were mentioned in the assignment.
Retrieve the kubeconfig of the cluster such that with one manual link the user can directly access the cluster from their own terminal.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,10 +16,78 @@ docker login ghcr.io
 Then you can deploy the application by running:
 
 ```
-docker compose up
```
Comment (RodinHaker) [inline README.md:20]: This should be `docker compose`.
`docker-compose` (note the dash) is the older version.

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+# Configure the VM
+Vagrant.configure("2") do |config|
+    # The box and version used
+    config.vm.box = "bento/ubuntu-24.04"
+    config.vm.box_version = "202404.26.0"
+
+    # Define the controller node
+    config.vm.define "controller" do |controller|
+        controller.vm.hostname = "controller"
+        controller.vm.network "private_network", ip: nodes_ip + "#{start_device}"
```
Comment (RodinHaker) [inline Vagrantfile]: The ip part of this could be `"#{nodes_ip}#{start_device}"`

File: Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            vb.memory = controller_mem
+            vb.cpus = controller_cores
+        end
+    end
+
+    # Define the worker nodes
+    (1..workers).each do |i|
+        config.vm.define "node#{i}" do |worker|
+            worker.vm.hostname = "node#{i}"
+            worker.vm.network "private_network", ip: nodes_ip + "#{start_device + i}"
```
Comment (RodinHaker) [inline Vagrantfile]: Same here `"#{nodes_ip}#{start_device + i}"`

File: Vagrantfile [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        end
+    end
+
+    # Set up the Ansible playbooks
+    config.vm.provision :ansible do |a|
+        a.compatibility_mode = "2.0"
+        a.playbook = "ansible/provisioning.yml"
+        a.inventory_path = "ansible/inventory.cfg"
+        a.extra_vars = {
+            "controller_ip" => nodes_ip + "#{start_device}"
```
Comment (RodinHaker) [inline Vagrantfile]: And here `"#{nodes_ip}#{start_device}"`

**Review by RodinHaker — APPROVED** [review]
I do not have much to say on this, looks solid!

File: README.md (hunk 1 above)
Comment (tpenning) [reply README.md:20]: Oh my bad I didn't know that and therefore changed it, I'll change it back then

File: Vagrantfile (hunk 2 above)
Comment (tpenning) [reply Vagrantfile]: Improved this

File: Vagrantfile (hunk 4 above)
Comment (tpenning) [reply Vagrantfile]: Improved this

File: Vagrantfile (hunk 3 above)
Comment (tpenning) [reply Vagrantfile]: Improved this

**Pull request merged**